src/zenml/orchestrators/utils.py

```diff
@@ -50,6 +56,44 @@ def create_tfx_pipeline(
     )
 
 
+def get_cache_status(
```

Comment:
Might be worth writing a unit few tests for this function to make sure it gets out the right thing

Reply:
I probably would have to mock up lots of pieces to make this work fast: the Repository, the metadata_store, specific pipelines and steps. Do you see any way to do this unit test without doing all of that? (Also maybe @schustmi you did some of this mocking already of certain components given that you recently wrote a bunch of tests across stack components...)

Reply:
Mhm yes certainly not an easy test, but we can maybe get away with not doing any mocking at all:
- Run the same pipeline (with cache enabled) twice inside a `clean_repo`
- Create an `execution_info` object with the right pipeline/run/step names
- Check if it returns `False` for the first run and `True` for the second one

Reply:
Ok so it'd be an integration test, then, and not a unit test. Yep that's easy to do.

Reply:
Actually I misspoke. Not an integration test, but somewhere in between. I saw how you've been using `clean_repo` in other tests and modelled something around that in my implementation.